### PR TITLE
Move earth sensor computations to core

### PR DESCRIPTION
**Pull request by Yash-10**
This PR tries to move computations related to earth sensors to `core`. 

Additionally, the conversion of variable names (in the extended ascii format) to printable ascii characters have been made for consistency.

Thanks!!

### CODE REVIEW

**Comment by Yash-10** [comment]
@astrojuanlu I think #1196 could get resolved here. Could you have a look?

The codecov/patch test didn't execute for some reason. I am trying to figure out the reason for this..

Thanks!

**Comment by astrojuanlu** [comment]
Indeed, #1196 got solved here! Do you know if it's just the whitespace or something else?

**Comment by Yash-10** [comment]
> Do you know if it's just the whitespace or something else?

I think it would not have to do with whitespace issues. I would try to find the reason behind this :)

**Comment by astrojuanlu** [comment]
I opened #1199 to discuss about variable names. But I'm +0 (meaning: not totally convinced, but slightly positive) on this change.

File: src/poliastro/earth/sensors.py [hunk 1]
```diff
@@ -1,110 +1,107 @@
...
 
+from poliastro.core.sensors import (
+    max_and_min_ground_range_with_specific_azimuth as max_and_min_ground_range_with_specific_azimuth_fast,
+    min_and_max_ground_range as min_and_max_ground_range_fast,
+)
+
 
-@u.quantity_input(h=u.km, η_fov=u.rad, η_center=u.rad, R=u.km)
-def min_and_max_ground_range(h, η_fov, η_center, R):
+@u.quantity_input(h=u.km, n_fov=u.rad, n_center=u.rad, R=u.km)
```
Comment (astrojuanlu) [inline src/poliastro/earth/sensors.py:9]: If we're doing this, let's not rename `η` to `n`, and use `eta` instead.

```suggestion
@u.quantity_input(h=u.km, eta_fov=u.rad, eta_center=u.rad, R=u.km)
```

(this applies everywhere)

File: src/poliastro/earth/sensors.py [hunk 2]
```diff
@@ -1,110 +1,107 @@
...
-    η_center: ~astropy.units.Quantity
+    n_center: ~astropy.units.Quantity
         Center boresight angle.
     R: ~astropy.units.Quantity
         Attractor equatorial radius
+
     Returns
     -------
-    Λ_min: ~astropy.units.Quantity
+    lambda_min: ~astropy.units.Quantity
```
Comment (astrojuanlu) [inline src/poliastro/earth/sensors.py:27]: For some reason, `~astropy.units.Quantity` does not always work:

Screenshot_2021-05-03 poliastro earth sensors — poliastro 0 15 dev0 documentation(1)

File: src/poliastro/earth/sensors.py [hunk 3]
```diff
@@ -1,110 +1,107 @@
...
         Latitude angle of the target point.
-    λ_tgt: ~astropy.units.Quantity
+    lambda_tgt: ~astropy.units.Quantity
         Longitude angle of the target point.
+
     Raises
     ------
     ValueError
-        This formula always gives the answer for the short way to the target ot the acute angle,:math:`{\beta}`,
+        This formula always gives the answer for the short way to the target ot the acute angle, beta,
```
Comment (astrojuanlu) [inline src/poliastro/earth/sensors.py:87]: Why?

File: tests/tests_earth/test_sensors.py [hunk 4]
```diff
@@ -96,17 +98,17 @@ def test_exception_max_and_min_ground_range_with_specific_azimuth(
...
 ):
 
     R = Earth.R.to(u.km)
     with pytest.raises(ValueError) as excinfo:
         max_and_min_ground_range_with_specific_azimuth(
-            h, n_center, n_fov, β, φ_nadir, λ_nadir, R
+            h, n_center, n_fov, beta, phi_nadir, lambda_nadir, R
         )
-    assert "β must be between 0º and 180º" in excinfo.exconly()
+    assert "beta must be between 0º and 180º" in excinfo.exconly()
```
Comment (astrojuanlu) [inline tests/tests_earth/test_sensors.py:114]: +1 on being consistent between variable names and error messages.

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Left some comments, but apart from that, this looks good to me!

This is on the verge of not being merged though, since I am preparing the beta release now. So if we could make a smaller PR just fixing the docstring issues, that would be great.

**Comment by Yash-10** [comment]
> This is on the verge of not being merged though, since I am preparing the beta release now.

Thanks for the suggestions, I would make the required changes and we shall look at it after the release!

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Thanks again for this @Yash-10 ! I have recently renamed one of the functions, could you please rebase this on top of current `main`?

**Comment by Yash-10** [comment]
Sorry @astrojuanlu for not keeping this branch up to date. While I rebase, could you let me know from the discussions in #1199, whether we have decided to keep the variable names as they are in current `main` or the one I had changed?

Thanks!

**Comment by Yash-10** [comment]
Since it was decided to use ascii variable names at the high-level and allow using non-ascii at the low level, I have tried to change it accordingly.

The `codeclimate` failure seems a bit tricky. It says `ground_range_diff_at_azimuth` has 10 arguments but infact it has only 7 arguments. I wonder if it has something to do with the function being both jitted and uses non-ascii variables?

**Comment by astrojuanlu** [comment]
@Yash-10 sorry, don't pay much attention to CodeClimate, we are not strict about it (in fact, we look at it very rarely). You can revert your last commit it you want, I'll look at your PR soon.

**Comment by Yash-10** [comment]
Thanks @astrojuanlu I reverted the recent commit.

It was interesting to note that changing variable names to ascii in core had removed the codeclimate issues. If there is a need, we could probably think of configuring `codeclimate.yml` to ignore such issues if non-ascii variables are used in core anywhere..

**Comment by astrojuanlu** [comment]
@Yash-10 could you please rebase this one on top of `main` after we merge #1191? :)

**Review by astrojuanlu — APPROVED** [review]
This is good to go!

**Comment by astrojuanlu** [comment]
Automerge worked! Nice 😎

**Comment by Yash-10** [comment]
Its really great! Thanks for merging!

**Pull request merged**